src/megatron/energon/wrappers/shuffle_buffer_dataset.py

```diff
@@ -51,8 +31,12 @@ def __init__(
         """Create a shuffle buffer for the dataset."""
         super().__init__(dataset, worker_config=worker_config)
         self.size = size
+        self.reset_state_own()
+
+    def reset_state_own(self) -> None:
         self._worker_rng = WorkerRng(self.worker_config)
-        self._active_buffer = SavableSampleBuffer(dataset, worker_config=worker_config)
+        # TODO: Former reset was via restore_state(None). Is this ok too?
```

Comment:
I guess it would be nice to have this pulled through every class, that the savables are mostly instantiated in the init? And the respective savable will call `.reset_state_*` on the inner types?